java/ql/src/experimental/Security/CWE/CWE-1004/SensitiveCookieNotHttpOnly.ql

```diff
@@ -0,0 +1,217 @@
+/**
+ * @name Sensitive cookies without the HttpOnly response header set
+ * @description Sensitive cookies without the 'HttpOnly' flag set leaves session cookies vulnerable to
+ *              an XSS attack. This query checks whether 'HttpOnly' is not set in a Java Cookie object
+ *              or the 'Set-Cookie' HTTP header.
+ * @kind path-problem
+ * @id java/sensitive-cookie-not-httponly
+ * @tags security
+ *       external/cwe/cwe-1004
+ */
+
+import java
```

Comment:
```suggestion
/*
 * Sketch of the structure of this query: we track cookie names that appear to be sensitive (e.g. `session` or `token`)
 * to a `ServletResponse.addHeader(...)` or `.addCookie(...)` method that does not set the `httpOnly` flag.
 * Subsidiary configurations `MatchesHttpOnlyConfiguration` and `SetHttpOnlyInCookieConfiguration` are used
 * to establish when the `httpOnly` flag is likely to have been set, before configuration `MissingHttpOnlyConfiguration`
 * establishes that a non-`httpOnly` cookie has a sensitive-seeming name.
 */

import java
```

Reply:
Thanks for the detailed description. I've added it.